src/coreclr/scripts/superpmi.py

```diff
@@ -867,6 +874,69 @@ async def run_pmi(print_prefix, assembly, self):
                 helper.run_to_completion(run_pmi, self)
 
                 os.environ.update(old_env)
+            ################################################################################################ end of "self.coreclr_args.pmi is True"
+
+            if self.coreclr_args.crossgen is True:
+                async def run_crossgen(print_prefix, assembly, self):
+                    """ Run crossgen over all dlls
+                    """
+
+                    root_crossgen_output_filename = make_safe_filename("crossgen_" + assembly) + ".out.dll"
+                    crossgen_output_assembly_filename = os.path.join(self.temp_location, root_crossgen_output_filename)
+                    try:
+                        if os.path.exists(crossgen_output_assembly_filename):
+                            os.remove(crossgen_output_assembly_filename)
+                    except OSError as ose:
+                        if "[WinError 32] The process cannot access the file because it is being used by another " \
+                           "process:" in format(ose):
+                            logging.warning("Skipping file %s. Got error: %s".format(root_output_filename, format(ose)))
+                            return
+                        else:
+                            raise ose
+
+                    command = [self.crossgen_tool, "/Platform_Assemblies_Paths", self.core_root, "/in", assembly, "/out", crossgen_output_assembly_filename]
+                    command_string = " ".join(command)
+                    logging.debug("%s%s", print_prefix, command_string)
+
+                    # Save the stdout and stderr to files, so we can see if crossgen wrote any interesting messages.
+                    # Use the name of the assembly as the basename of the file. mkstemp() will ensure the file
+                    # is unique.
+                    root_output_filename = make_safe_filename("crossgen_" + assembly + "_")
+                    try:
+                        stdout_file_handle, stdout_filepath = tempfile.mkstemp(suffix=".stdout", prefix=root_output_filename, dir=self.temp_location)
+                        stderr_file_handle, stderr_filepath = tempfile.mkstemp(suffix=".stderr", prefix=root_output_filename, dir=self.temp_location)
+
+                        proc = await asyncio.create_subprocess_shell(
```

Comment:
There is common code is pmi and crossgen that you can extract in a common function?

Reply:
There is common code, but I didn't want to worry about extracting a common function right now